tests/components/test_datadog.py

```diff
@@ -0,0 +1,187 @@
+"""The tests for the Datadog component."""
+from unittest import mock
+import unittest
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    EVENT_LOGBOOK_ENTRY,
+    EVENT_STATE_CHANGED,
+    STATE_OFF,
+    STATE_ON
+)
+from homeassistant.setup import setup_component
+import homeassistant.components.datadog as datadog
+import homeassistant.core as ha
+
+from tests.common import get_test_home_assistant
+
+
+class TestDatadog(unittest.TestCase):
+    """Test the Datadog component."""
+
+    def setUp(self):  # pylint: disable=invalid-name
+        """Setup things to be run when tests are started."""
+        self.hass = get_test_home_assistant()
+
+    def tearDown(self):  # pylint: disable=invalid-name
+        """Stop everything that was started."""
+        self.hass.stop()
+
+    def test_invalid_config(self):
+        """Test configuration with defaults."""
+        config = {
+            'datadog': {
+                'host1': 'host1',
+            }
+        }
+
+        with self.assertRaises(vol.Invalid):
```

Comment:
Good call. I grabbed this from the [statsd](https://github.com/home-assistant/home-assistant/blob/dev/tests/components/test_statsd.py#L26-L37) component, which I used for the basis of my component.

Would you accept a commit that improves the test there as well?